### ISSUE TITLE
Sleuth breaks the span continuity by generating multiple spans for for same request

### ISSUE THREAD

**Issue by nnanda2016** [spring-cloud/spring-cloud-sleuth#1110]
Sample Repo: https://github.com/nnanda2016/springboot-sleuth-couchbase-issue
Spring Boot version: `2.0.4.RELEASE`
Spring cloud release train: `Finchley.SR1`

For the lifecyle of a single request, sleuth is generating multiple spans (different traceId/spanId pairs). The `README.MD` file shows how to replicate the issue. This breaks the transaction traceability which is why we need Sleuth.

**Comment by marcingrzejszczak** [comment]
Can you make the sample simpler, for example without setting any versions manually?

**Comment by nnanda2016** [comment]
Hi Marcin, I noticed this issue only in following scenarios
- if the application has multiple layers (like DAO, Service, web controller/handler etc)
- if the application is making remote calls.

I know in the past you have always asked for simpler samples and I had kept my other samples simpler. But this one, I tried reproducing in a simpler app, but could not. Hence, this complexity.

**Comment by nnanda2016** [comment]
Hi Marcin, I have simplified the sample app. Please check if that is helpful.

https://github.com/nnanda2016/springboot-sleuth-couchbase-issue

**Comment by marcingrzejszczak** [comment]
Sorry @nnanda2016 but this example is terribly complex. You set plenty of versions by hand. Please try to use the latest versions and verify if things work better. If not please, create the simplest possible scenario.

**Comment by nnanda2016** [comment]
@marcingrzejszczak From Spring Cloud Sleuth perspective, I have used `Finchley.SR1`. I am not sure if I understood your mention of `plenty of versions by hand`. From Spring Cloud perspective, `open-tracing` and `reactor-core` are the only libraries for which I am hardcoding the version. Others should not bother you. Open Tracing was required by Couchbase SDK and without this I was getting class not found exception while running the app.

Anyways, since this is closed, I have opened a new issue with simplified sample app.
